x-pack/plugins/rule_registry/common/assets/field_maps/technical_rule_field_map.test.ts

```diff
@@ -198,6 +198,31 @@ it('matches snapshot', () => {
         "required": false,
         "type": "keyword",
       },
+      "kibana.alert.throttle.count": Object {
+        "array": false,
+        "required": false,
+        "type": "long",
+      },
+      "kibana.alert.throttle.end": Object {
+        "array": false,
+        "required": false,
+        "type": "date",
+      },
+      "kibana.alert.throttle.fields": Object {
+        "array": true,
+        "required": false,
+        "type": "keyword",
+      },
+      "kibana.alert.throttle.start": Object {
+        "array": false,
+        "required": false,
+        "type": "date",
+      },
+      "kibana.alert.throttle.values": Object {
+        "array": true,
+        "required": false,
+        "type": "keyword",
+      },
```

Comment:
I initially went with this approach to avoid having arbitrary field names inside `throttled_fields`, but we could probably make that work with the `flattened` field type. I figured since we won't be able to fully map the field values in `throttled_fields` that querying the top-level versions of those fields would be preferred, as any ECS fields will be correctly mapped. E.g. `host.ip` would be mapped as IP, but `throttle_summary.throttled_fields.host.ip` would at best be usable as a keyword if we map `throttled_fields` as `flattened`.

I think looking at it again I like the idea of keeping the field-value pairs in an object like `throttled_fields` better than separate arrays.

Reply:
> we could probably make that work with the flattened field type

@marshallmain Yep, my vote definitely goes for the flattened field type and for a map of key-value pairs instead of two arrays (`kibana.alert.suppression.terms.field` and `kibana.alert.suppression.terms.value`)

Reply:
I tried implementing it with a flattened field and ran into problems handling the alerts in timeline - since timeline destroys the original structure of the document, we can't simply grab the `throttled_fields` object and iterate over the keys. I switched to the approach @mikecote suggested instead, which is the same way we implement key-value pair storage for `threshold_result` in threshold rules.